pkg/kubelet/cm/topologymanager/topology_manager.go

```diff
@@ -43,10 +44,20 @@ const (
 )
 
 // TopologyAffinityError represents an resource alignment error
-type TopologyAffinityError struct{}
+type TopologyAffinityError struct {
+	ContainerName string
+	Hint          string
+}
 
 func (e TopologyAffinityError) Error() string {
-	return "Resources cannot be allocated with Topology locality"
+	msg := "Resources cannot be allocated with Topology locality"
```

Comment:
In general, yes. In the specific context of generating the message for this error, I think the trivial string concatenation is the sweet spot because we are expected to do this rarely, and the code is very trivial to read.